### ISSUE TITLE
Unable to infer types from `uom` crate & false positive `E0308` with `uom` crate types

### ISSUE THREAD

**Issue by chamoretto** [intellij-rust/intellij-rust#8800]
## Environment

* **IntelliJ Rust plugin version:** 0.4.169.4562-221-nightly
* **Rust toolchain version:** 1.61.0-nightly (76d770ac2 2022-04-02) x86_64-pc-windows-msvc
* **IDE name and version:** CLion 2022.1 (CL-221.5080.224)
* **Operating system:** Windows 10 10.0
* **Macro expansion engine:** new
* **Name resolution engine:** new
* **Additional experimental features:** org.rust.macros.proc

![image](https://user-images.githubusercontent.com/28485306/165939516-d3d1d7fb-7790-4209-95db-ab09d21ec40f.png)

## Problem description

Plugin is unable to infer types with `uom` crate, [but it should, previous issue had been resolved](https://github.com/intellij-rust/intellij-rust/issues/4945#issuecomment-866181595)
![image](https://user-images.githubusercontent.com/28485306/165939924-a9faf107-ba24-4c05-ba03-bdc966d315db.png)

And also following code and crate generates false-positive `E0308` (code compiles successfully, look at the screenshot)
![image](https://user-images.githubusercontent.com/28485306/165940146-115f5474-242e-4228-9adf-90f2ae763468.png)

## Steps to reproduce
Add following into your `Cargo.toml`
```
[dependencies]
uom = "0.32.0"
```

Code for test:
```rust
use uom::si::f32::*;
use uom::si::length::kilometer;
use uom::si::time::second;

#[test]
fn test_smth() {
    // TODO: test uom
    use uom::si::f32::Information;
    use uom::si::information::byte;
    use uom::si::ratio::*;

    let b = Information::new::<byte>(255.0);
    let a = b + b;

    let length = Length::new::<kilometer>(5.0);
    let time = Time::new::<second>(15.0);
    let velocity = length / time;
    let _acceleration = calc_acceleration(velocity, time);

    println!("{:?}, {:?}", a, b);
}

fn calc_acceleration(velocity: Velocity, time: Time) -> Acceleration {
    velocity / time
}
```

**Comment by chamoretto** [comment]
Oh, i've missed disabled feature `org.rust.cargo.evaluate.build.scripts`, confused this options with the old buggy `org.rust.cargo.fetch.out.dir`, sorry.
